spec/workflow/intrinsic_function/parser_spec.rb

```diff
@@ -0,0 +1,213 @@
+# Ignore percent-delimeter cop because our test Strings can have all kinds of
+#   characters and using `%q||` keeps the tests consistent, even if it's not
+#   needed.
+# rubocop:disable Style/PercentLiteralDelimiters, Style/RedundantPercentQ
+
+require 'parslet/rig/rspec'
+
+RSpec.describe Floe::Workflow::IntrinsicFunction::Parser do
+  describe "spaces" do
+    subject { described_class.new.spaces }
+
+    it do
+      expect(subject).to parse(" ")
```

Comment:
Is `expect().to parse()` just checking that it doesn't raise an exception?  Is there a parse result we could be checking?

Reply:
This tests the parser directly and not the transformer, so yeah, more or less.  The results from the parser are just the tree, which could be a useful set of tests especially when parsing something more complicated like nested functions, but here I just wanted validation of good/bad cases.

Here's some example results:

```ruby
$ bin/console
irb(main):001:0> Floe::Workflow::IntrinsicFunction::Parser.new.spaces.parse(" ")
=> " "@0
irb(main):002:0> Floe::Workflow::IntrinsicFunction::Parser.new.number.parse("123.456")
=> {:number=>"123.456"@0}
irb(main):003:0> Floe::Workflow::IntrinsicFunction::Parser.new.number.parse("xxx")
/Users/jfrey/.gem/ruby/3.1.5/gems/parslet-2.0.0/lib/parslet/cause.rb:70:in `raise': Failed to match sequence ('-'? ('0' / [1-9] DIGIT{0, }) ('.' DIGIT{1, })? ([eE] ('+' / '-')? DIGIT{1, })?) at line 1 char 1. (Parslet::ParseFailed)
	from /Users/jfrey/.gem/ruby/3.1.5/gems/parslet-2.0.0/lib/parslet/atoms/base.rb:49:in `parse'
	from (irb):3:in `<main>'
	from bin/console:17:in `<main>'